components/page/DevPageProjects.tsx

```diff
@@ -0,0 +1,296 @@
+import styles from './devpage.module.css';
+import JSZip from 'jszip';
+import { saveAs } from 'file-saver';
+
+import {
+  Card,
+  CardHeader,
+  CardTitle,
+  CardBody,
+  CardExpandableContent,
+  Button,
+} from '@patternfly/react-core';
+import { useState } from 'react';
+
+/**
+ * Representation of a Git source for  a starter project
+ *
+ * @param checkoutFrom - object with optional remote and revision keys
+ *                       and specifies where project should be checked out
+ * @param remote - remote name in remotes that should be used as the init and download url
+ * @param revision - revision to checkout from
+ * @param remotes - remotes map; should be init. to git project
+ */
+interface Git {
+  checkoutFrom?: {
+    remote?: string;
+    revision?: string;
+  };
+  remotes: Record<string, string>;
+}
+
+/**
+ * Representation of a single starter project
+ *
+ * @remarks
+ *  For more details, check out the devfile API reference at https://docs.devfile.io/devfile/2.1.0/user-guide/api-reference.html
+ *
+ * @param name - name of starter project
+ * @param description - description of starter project
+ * @param git - git source of starter project
+ * @param zip - zip source of start project with key location
+ * @param location - zip project's source location address
+ * @param subDir - subdirectory of zip to be used as the root for the starter project
+ */
+interface Project {
+  name: string;
+  description?: string;
+  git?: Git;
+  zip?: {
+    location: string;
+  };
+  subDir?: string;
+}
+
+/**
+ * props for {@link DevPageProjects}
+ *
+ * @param starterProjects - list of starter projects
+ */
+interface Props {
+  starterProjects: Project[];
+}
+
+/**
+ * download subdirectory from root folder
+ *
+ *
+ * @param url - zip url
+ * @param subdirectory - name of subdirectory to extract from zip
+ *
+ * @throws Error
+ *    thrown if error in download
+ */
+async function downloadSubdirectory(url: string, subdirectory: string) {
+  const data = {
+    url,
+    subdirectory,
+  };
+  const response = await fetch('/api/download-subdirectory', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(data),
+  });
+
+  const status = response.status;
+  if (status !== 200) {
+    const errorJson = await response.json();
+    console.log(errorJson);
+    throw new Error(errorJson);
+  }
+
+  const base64string = await response.text();
+  const zip = await JSZip.loadAsync(base64string, { base64: true });
+
+  await zip.generateAsync({ type: 'blob' }).then(
+    function (blob) {
+      saveAs(blob, subdirectory + '.zip');
+    },
+    function (err) {
+      jQuery('#blob').text(err);
```

Comment:
should this be `throw new Error`; not sure how jquery works in our project

Reply:
resolved